extension/styles.css

```diff
@@ -342,4 +347,118 @@ input[type="checkbox"] {
   background-color: #e8eaed;
   color: var(--primary-color);
   text-decoration: none;
-} 
\ No newline at end of file
+} 
+
+/* Dark mode class */
+body.dark-mode {
+  background-color: var(--dark-bg-color);
+  color: var(--dark-text-color);
+}
+
+body.dark-mode .container {
+  background-color: var(--dark-bg-color);
+}
+
+body.dark-mode .header h1 {
+  color: var(--primary-color);
+}
+
+body.dark-mode .tagline {
+  color: var(--dark-dark-gray);
+}
+
+body.dark-mode .preview-container {
+  background-color: var(--dark-light-gray);
+}
+
+body.dark-mode .preview-content {
+  background-color: var(--dark-bg-color);
+  border-color: var(--dark-border-color);
+  color: var(--dark-text-color);
+}
+
+body.dark-mode .settings-toggle-btn {
+  background-color: var(--dark-light-gray);
+  border-color: var(--dark-border-color);
+  color: var(--dark-text-color);
+}
+
+body.dark-mode .settings-toggle-btn:hover {
+  background-color: #3d3d3d;
+}
+
+body.dark-mode .settings-container {
+  background-color: var(--dark-bg-color);
+  border-color: var(--dark-border-color);
+}
+
+body.dark-mode .shortcut-info {
+  background-color: var(--dark-light-gray);
+}
+
+body.dark-mode .shortcut-row code {
+  background-color: var(--dark-bg-color);
+  border-color: var(--dark-border-color);
+  color: var(--dark-text-color);
+}
+
+body.dark-mode .shortcut-customize {
+  color: var(--dark-dark-gray);
+}
+
+body.dark-mode .shortcut-customize code {
+  background-color: var(--dark-light-gray);
+  color: var(--dark-text-color);
+}
+
+/* Dark mode toggle button styles */
```

Comment:
Smart approach to icon switching using CSS display properties rather than JavaScript DOM manipulation. This keeps the visual logic in CSS where it belongs and is more performant! 👍